tests/forte/data/ontology/ontology_code_generator_test.py

```diff
@@ -175,6 +175,130 @@ def test_include_and_exclude_init(self):
             exp_files = [file for file in exp_files if file != corrupted_path]
             self.assertEqual(gen_files, exp_files)
 
+    def test_with_init(self):
+        json_file_path = os.path.join(
+            self.spec_dir, "example_import_ontology.json"
+        )
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_filename = _get_temp_filename(json_file_path, temp_dir)
+            # Test with include_init = True, with_init=True
+            folder_path = self.generator.generate(
+                temp_filename,
+                temp_dir,
+                is_dry_run=False,
+                include_init=True,
+                with_init=True,
+            )
+            gen_files = sorted(utils.get_generated_files_in_dir(folder_path))
+
+            # Assert the generated python files
+            exp_file_path = [
+                "ft/__init__",
+                "ft/onto/__init__",
+                "ft/onto/example_import_ontology",
+            ]
+            exp_files = sorted(
+                [
+                    f"{os.path.join(folder_path, file)}.py"
+                    for file in exp_file_path
+                ]
+            )
+            self.assertEqual(gen_files, exp_files)
+
+            # Now, corrupt one of the init files
+            corrupted_path = os.path.join(folder_path, "ft/__init__.py")
+            with open(corrupted_path, "w") as f:
+                f.write("# ***corrupted file***\n")
+
+            # Re-generate using include_init = False
+            self.generator = OntologyCodeGenerator()
+            folder_path = self.generator.generate(
+                temp_filename,
+                folder_path,
+                is_dry_run=False,
+                include_init=False,
+                with_init=True,
+            )
+            gen_files = sorted(utils.get_generated_files_in_dir(folder_path))
+
+            # Assert the generated python files after removing the corrupted
+            # file which should not have been regenerated
+            exp_files = [file for file in exp_files if file != corrupted_path]
+            self.assertEqual(gen_files, exp_files)
+
+    def test_without_init(self):
+        json_file_path = os.path.join(
+            self.spec_dir, "example_import_ontology.json"
+        )
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_filename = _get_temp_filename(json_file_path, temp_dir)
+            # Test with include_init = True, with_init=False
+            folder_path = self.generator.generate(
+                temp_filename,
+                temp_dir,
+                is_dry_run=False,
+                include_init=True,
+                with_init=False,
+            )
+            gen_files = sorted(utils.get_generated_files_in_dir(folder_path))
+
+            # Assert the generated python files
+            exp_file_path = [
+                "ft/__init__",
```

Comment:
`__init__` file is still here? I thought we want to not generate them if the variable is False

Reply:
Could you explain why when `use_name_space_packaging` is true, we still have `ft/__init__.py`? And please write write `ft/__init__.py`. Do not append ".py" later, this adds complexity in understanding the logic